pkg/operators/cli/clioperator.go

```diff
@@ -241,15 +245,34 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 			// TODO: compatiblity for now: add all; remove me later on and use the commented version above
 			opts = []json.Option{json.WithShowAll(true)}
 
+			if o.mode == ModeJSONPretty {
+				opts = append(opts, json.WithPretty(true, "  "))
+			}
+
 			jsonFormatter, err := json.New(ds, opts...)
 			if err != nil {
 				return fmt.Errorf("initializing JSON formatter: %w", err)
 			}
 
-			ds.Subscribe(func(ds datasource.DataSource, data datasource.Data) error {
+			df := func(ds datasource.DataSource, data datasource.Data) error {
 				fmt.Println(string(jsonFormatter.Marshal(data)))
 				return nil
-			}, Priority)
+			}
+
+			if o.mode == ModeYAML {
+				// For the time being, this uses a slow approach to marshal to YAML, by first
+				// converting to JSON and then to YAML. This should get a dedicated formatter sooner or later.
+				df = func(ds datasource.DataSource, data datasource.Data) error {
+					yml, err := yaml.JSONToYAML(jsonFormatter.Marshal(data))
+					if err != nil {
+						return fmt.Errorf("serializing yaml: %w", err)
+					}
+					fmt.Println("---")
+					fmt.Println(string(yml))
```

Comment:
`yml` already contains a final newline:

```suggestion
					fmt.Print(string(yml))
```

Current output:

```bash
INFO[0000] Experimental features enabled
---
ino: 39847
k8s:
  container: ""
  hostnetwork: false
  namespace: ""
  node: ""
  pod: ""
netns: 4026532260
runtime:
  containerId: 7bae168baa38838a73e1a4b384c526af717cfd2b45197bd2833bf4863466cc9d
  containerImageDigest: ""
  containerImageName: nginx:1.21
  containerName: shell01
  runtimeName: docker
state: 10

---
ino: 39846
k8s:
  container: ""
  hostnetwork: false
  namespace: ""
  node: ""
  pod: ""
netns: 4026532260
runtime:
  containerId: 7bae168baa38838a73e1a4b384c526af717cfd2b45197bd2833bf4863466cc9d
  containerImageDigest: ""
  containerImageName: nginx:1.21
  containerName: shell01
  runtimeName: docker
state: 10

```

After this change:

```bash
INFO[0000] Experimental features enabled
---
ino: 39847
k8s:
  container: ""
  hostnetwork: false
  namespace: ""
  node: ""
  pod: ""
netns: 4026532260
runtime:
  containerId: 7bae168baa38838a73e1a4b384c526af717cfd2b45197bd2833bf4863466cc9d
  containerImageDigest: ""
  containerImageName: nginx:1.21
  containerName: shell01
  runtimeName: docker
state: 10
---
ino: 39846
k8s:
  container: ""
  hostnetwork: false
  namespace: ""
  node: ""
  pod: ""
netns: 4026532260
runtime:
  containerId: 7bae168baa38838a73e1a4b384c526af717cfd2b45197bd2833bf4863466cc9d
  containerImageDigest: ""
  containerImageName: nginx:1.21
  containerName: shell01
  runtimeName: docker
state: 10
```